debezium-server/debezium-server-core/src/main/java/io/debezium/server/storage/JdbcDatabaseHistory.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.server.storage;
+
+import com.google.common.base.Splitter;
+import io.debezium.annotation.ThreadSafe;
+import io.debezium.config.Configuration;
+import io.debezium.config.Field;
+import io.debezium.document.DocumentReader;
+import io.debezium.document.DocumentWriter;
+import io.debezium.relational.history.*;
+import io.debezium.util.Collect;
+import io.debezium.util.FunctionalReadWriteLock;
+import org.apache.kafka.connect.errors.ConnectException;
+
+import java.io.IOException;
+import java.sql.*;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Consumer;
+
+/**
+ * A {@link DatabaseHistory} implementation that stores the schema history in a local file.
+ *
+ * @author Randall Hauch
+ */
+@ThreadSafe
+public final class JdbcDatabaseHistory extends AbstractDatabaseHistory {
+
+    public static final String CONFIGURATION_FIELD_PREFIX_STRING = "offset.storage.";
+    public static final Field JDBC_URI = Field.create(CONFIGURATION_FIELD_PREFIX_STRING + "jdbc.uri")
+            .withDescription("Uri of the database which will be used to record the database history")
+            .withValidation(Field::isRequired);
+    public static final Field JDBC_USER = Field.create(CONFIGURATION_FIELD_PREFIX_STRING + "jdbc.user")
+            .withDescription("Username of the database which will be used to record the database history")
+            .withValidation(Field::isRequired);
+    public static final Field JDBC_PASSWORD = Field.create(CONFIGURATION_FIELD_PREFIX_STRING + "jdbc.password")
+            .withDescription("Password of the database which will be used to record the database history")
+            .withValidation(Field::isRequired);
+    public static final String DATABASE_HISTORY_TABLE_NAME = "debezium_database_history";
+    public static final String DATABASE_HISTORY_TABLE_DDL = "CREATE TABLE " + DATABASE_HISTORY_TABLE_NAME +
+            "(" +
+            "history_insert_epoch_ms VARCHAR(255) NOT NULL," +
+            "history_data_part1 VARCHAR(65000)," +
```

Comment:
Instead of having the three columns, how about rather flowing over large entries into multiple rows. That'd be the more reliable approach.